specification/dartLangSpec.tex

```diff
@@ -11641,6 +11765,246 @@ \subsection{Assignable Expressions}
 Evaluation of an assignable expression of the form \code{\SUPER{}[$e_2$]} is equivalent to evaluation of the method invocation \code{\SUPER{}.[]($e_2$)}.
 
 
+\subsection{Lexical Lookup}
+\LMLabel{lexicalLookup}
+
+\LMHash{}%
+This section specifies how to look up a name
+based on the enclosing lexical scopes.
+This is known as a \Index{lexical lookup}.
+When \id{} is an identifier, it may look up a name $n$ of the form \id{}
+as well as of the form \code{\id=}.
+
+\LMHash{}%
+A lexical lookup does not succeed or fail.
```

Comment:
That's meaningless.
Does it "not *just* succeed or fail"?

Generally don't waste words saying what *isn't* true. Especially, don't *start* with it.

Reply:
But it doesn't fail at all. It _always_ delivers a result. (So it's also meaningless to say that it 'succeeds'.)

A compile-time error may occur during the lexical lookup, but we usually do not specify propagation of errors (for instance, when specifying `e++` we don't say 'if a compile-time error occurred during the determination of the static type of `e` then `e++` is also a compile-time error').

I reorganized the text such that it talks about the positive outcome first, and then explains why we don't have to mention the case `if the lexical lookup failed` in the locations where such a lookup is performed.

Reply:
With the rewrite of the whole paragraph, I believe this issue has been resolved.